commit 1076a25b43931dafaa5744fb71cd7682efb346ff
Author: Alexander Potapenko <glider@google.com>
Date:   Wed Jul 19 13:52:50 2017 +0200

    DO-NOT-SUBMIT: Clang flags

diff --git a/Makefile b/Makefile
index 853ae9179af9..22893282ee4c 100644
--- a/Makefile
+++ b/Makefile
@@ -629,7 +629,9 @@ ARCH_AFLAGS :=
 ARCH_CFLAGS :=
 include arch/$(SRCARCH)/Makefile
 
+ifneq ($(cc-name),clang)
 KBUILD_CFLAGS	+= $(call cc-option,-fno-delete-null-pointer-checks,)
+endif
 KBUILD_CFLAGS	+= $(call cc-disable-warning,frame-address,)
 
 ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
